src/posthog-core.js

```diff
@@ -620,6 +612,15 @@ PostHogLib.prototype.capture = addOptOutCheckPostHogLib(function (event_name, pr
         properties: this._calculate_event_properties(event_name, properties, start_timestamp),
     }
 
+    if (event_name === '$pageview' && this.get_config('_capture_performance')) {
```

Comment:
In an ideal world, methods do as little as possible - mixing business logic with capturing here is messy.

There is a call to `_calculate_event_properties` just above - extend that.